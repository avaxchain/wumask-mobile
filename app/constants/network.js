export const MAINNET = 'mainnet';
export const AVALANCHE = 'avalanche';
export const KOVAN = 'kovan';
export const RINKEBY = 'rinkeby';
export const GOERLI = 'goerli';
export const RPC = 'rpc';
