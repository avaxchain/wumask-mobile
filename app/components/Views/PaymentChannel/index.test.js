import React from 'react';
import { shallow } from 'enzyme';
import PaymentChannel from './';
import configureMockStore from 'redux-mock-store';
import { AVALANCHE } from '../../../constants/network';

const mockStore = configureMockStore();

describe('PaymentChannel', () => {
	it('should render correctly', () => {
		const initialState = {
			settings: {
				paymentChannelsEnabled: true
			},
			engine: {
				backgroundState: {
					NetworkController: {
						provider: {
							type: AVALANCHE,
							ticker: 'AVAX'
						}
					},
					TransactionController: {
						internalTransactions: [],
						transactions: []
					},
					CurrencyRateController: {
						currentCurrency: 'USD',
						nativeCurrency: 'AVAX',
						conversionRate: 100
					},
					PreferencesController: {
						selectedAddress: '0x1'
					},
					TokenRatesController: {
						contractExchangeRates: {}
					}
				}
			}
		};

		const wrapper = shallow(<PaymentChannel />, {
			context: { store: mockStore(initialState) }
		});
		expect(wrapper).toMatchSnapshot();
	});
});
