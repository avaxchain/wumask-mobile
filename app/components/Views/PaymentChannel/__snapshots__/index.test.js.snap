// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentChannel should render correctly 1`] = `
<withNavigation(withNavigationFocus(PaymentChannel))
  contractExchangeRates={Object {}}
  conversionRate={100}
  currentCurrency="USD"
  internalTransactions={Array []}
  nativeCurrency="AVAX"
  paymentChannelsEnabled={true}
  provider={
    Object {
      "ticker": "AVAX",
      "type": "avalanche",
    }
  }
  selectedAddress="0x1"
  setPaymentChannelTransaction={[Function]}
  showAlert={[Function]}
  transactions={Array []}
/>
`;
